### PR TITLE
fix alloc overflow when 0 pages allocated 

### PR DESCRIPTION
**Pull request by grishasobol**
Add test for the case.

### CODE REVIEW

**Review by ukint-vs — APPROVED** [review]

File: gclient/tests/upload.rs [hunk 1]
```diff
@@ -61,16 +50,65 @@ async fn harmless_upload() -> Result<()> {
...
+    // By default, login as Alice, than re-login as Bob.
+    let api = GearApi::dev().await?.with("//Bob")?;
+
+    upload_programs_and_check(&api, codes).await?;
+
+    Ok(())
+}
+
+#[test]
+fn alloc_zero_pages() {
```
Comment (breathx) [inline gclient/tests/upload.rs:85]: Why so complicated?

**Review by breathx — COMMENTED** [review]
Where is the fix itself?

File: gclient/tests/upload.rs (hunk 1 above)
Comment (grishasobol) [reply gclient/tests/upload.rs:85]: Can you offer simpler solution? I suppose there is some ways to make tests with timeout better.

Comment (SabaunT) [inline gclient/tests/upload.rs:85]: I guess it could be run in an async context.

**Review by SabaunT — APPROVED** [review]

File: gclient/tests/upload.rs (hunk 1 above)
Comment (SabaunT) [reply gclient/tests/upload.rs:85]: either test can be re-written or `block_stopped` checking function

Comment (breathx) [reply gclient/tests/upload.rs:85]: with tokio::test, exactly

Comment (grishasobol) [reply gclient/tests/upload.rs]: Done

**Pull request merged**